src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/resourcecard/model/ResourceCardItem.kt

```diff
@@ -1,5 +1,45 @@
 package org.wycliffeassociates.otter.jvm.app.widgets.resourcecard.model
 
-data class ResourceCardItem(
-    val title: String
-)
+import io.reactivex.disposables.CompositeDisposable
+import javafx.beans.property.DoubleProperty
+import javafx.beans.property.SimpleDoubleProperty
+import org.wycliffeassociates.otter.common.data.workbook.AssociatedAudio
+import org.wycliffeassociates.otter.common.data.workbook.Resource
+import org.commonmark.parser.Parser
+import org.commonmark.renderer.text.TextContentRenderer
+
+data class ResourceCardItem(val resource: Resource, val onSelect: () -> Unit) {
+    val title: String = renderTitleAsPlainText()
+    private val disposables = CompositeDisposable()
+    val titleProgressProperty: DoubleProperty = resource.titleAudio.progressProperty()
+    val bodyProgressProperty: DoubleProperty? = resource.bodyAudio?.progressProperty()
+    val hasBodyAudio: Boolean = resource.bodyAudio != null
+
+    @Suppress("ProtectedInFinal", "Unused")
+    protected fun finalize() {
+        clearDisposables()
```

Comment:
could you try putting a print statement here and seeing if this actually executes? I'm kinda concerned about the subscribe reference not allowing this to be marked for GC?

Reply:
This does in fact execute when the ResourceGroupCardItems are removed from the observable list, even when the ResourceCardItems have subscriptions to `AssociatedAudio.selected`. Since this is the case (and I should've realized this sooner), I suppose I don't need any of the additional logic that I added to manually clear the subscriptions. However, this was only tested when the ResourceGroupCardItem list is cleared programmatically- I don't know how to test it if the list is not cleared, i.e. how do I just drop the reference to the list or the ViewModel?

Reply:
I just read through the resolved discussion and realized that what makes me uncomfortable here is that clearDisposables() needs to be tied to a lifecycle API, and as you both discussed, onDock/onUndock are good lifecycle places to attach this to. In the absence of one, you're attaching to the GC lifecycle whether consciously or unconsciously.

The CardItem here isn't exactly a framework object, and as such has no APIs to make use of- so this functionality would have to be exposed as a public method here. I'm going to look and see if ListView has some kind of lifecycle APIs to clean cells that may be more appropriate?